### PR TITLE
Send daily email listing pending users

### LINKED ISSUE #576: Send daily email that lists pending users
https://github.com/dandi/dandi-api/pull/568#discussion_r734640814

Can be a celery beat task that checks for any pending users and sends an email that lists each user + the URL to approve them.

### PR DESCRIPTION
**Pull request by mvandenburgh**
Closes #576

### CODE REVIEW

File: dandiapi/api/tasks.py [hunk 1]
```diff
@@ -161,3 +163,12 @@ def validate_version_metadata(version_id: int) -> None:
 @shared_task
 def delete_doi_task(doi: str) -> None:
     delete_doi(doi)
+
+
+@shared_task
+def send_pending_users_email() -> None:
+    """Send an email to admins listing users with status set to PENDING"""
```
Comment (dchiquito) [inline dandiapi/api/tasks.py:170]:
```suggestion
    """Send an email to admins listing users with status set to PENDING."""
```

File: dandiapi/api/mail.py [hunk 2]
```diff
@@ -153,3 +154,20 @@ def send_rejected_user_message(user: User, socialaccount: SocialAccount):
...
     with mail.get_connection() as connection:
         connection.send_messages(messages)
+
+
+def build_pending_users_message(users: Union[QuerySet, List[User]]):
+    render_context = {'users': users}
+    return build_message(
+        subject='DANDI: new user registrations to review',
+        message=render_to_string('api/mail/pending_users_message.txt', render_context),
+        to=['dandi@mit.edu'],
```
Comment (dchiquito) [inline dandiapi/api/mail.py:164]: Probably this should be a constant/setting at some point

File: dandiapi/api/tasks.py [hunk 3]
```diff
@@ -161,3 +163,12 @@ def validate_version_metadata(version_id: int) -> None:
...
+    pending_users = User.objects.select_related('metadata').filter(
```
Comment (dchiquito) [inline dandiapi/api/tasks.py:171]: IIRC `select_related` is only useful if you're going to use `user.metadata` somewhere, as it saves the DB from performing a separate `SELECT`. I don't think you actually do that anywhere though, all I see are accesses of `user.username` and `user.date_joined` in the template. Are those aliases for metadata fields?

**Comment by satra** [comment]
can we use `info@dandiarchive.org` instead of `dandi@mit.edu` to make it more generic?

File: dandiapi/api/tasks.py (hunk 3 above)
Comment (mvandenburgh) [reply dandiapi/api/tasks.py]: Oops, I think I was accessing `user.metadata` somewhere but then removed it. And `username` and `date_joined` are both fields directly on the `User` model, so yeah select_related isn't needed here

**Comment by mvandenburgh** [comment]
> can we use `info@dandiarchive.org` instead of `dandi@mit.edu` to make it more generic?

Maybe the admin email should be a Django setting/env var that is set via Terraform. @dchiquito what do you think?

**Comment by dchiquito** [comment]
Assuming that there is one email that it is appropriate to send all admin traffic to, then yes it should definitely be a setting. Since we generally want to keep the repository divorced from our deployment, this is a task like https://github.com/dandi/dandi-api/issues/286 that would be a good idea to clean up at some point.

**Comment by mvandenburgh** [comment]
> Assuming that there is one email that it is appropriate to send all admin traffic to, then yes it should definitely be a setting. Since we generally want to keep the repository divorced from our deployment, this is a task like #286 that would be a good idea to clean up at some point.

:+1: I filed #605 to track this so we can keep this PR well-scoped. In the meantime, I've hardcoded `info@dandiarchive.org` as the email for DANDI admins.

**Review by dchiquito — APPROVED** [review]

**Pull request merged**